pkg/pillar/cmd/zedrouter/patchenvelopes.go

```diff
@@ -255,37 +256,15 @@ func (pes *PatchEnvelopes) UpdateVolumeStatus(vs types.VolumeStatus, deleteVolum
 		}
 		pes.completedVolumes.Store(vs.VolumeID, vs)
 	}
-	pes.affectedByVolumeStatus(vs)
-}
-
-// affectedByVolumeStatus fills in envelopesToUpdate map marking PatchEnvelopes
-// which need to be updated
-func (pes *PatchEnvelopes) affectedByVolumeStatus(vs types.VolumeStatus) {
-	UUIDList := pes.envelopes.Keys()
-
-	for _, UUID := range UUIDList {
-		if pe, ok := pes.envelopes.Load(UUID); ok {
-			for _, volRef := range pe.VolumeRefs {
-				volUUID, err := uuid.FromString(volRef.ImageID)
-				if err != nil {
-					pes.log.Errorf("Failed to compose volUUID from string %v", err)
-					continue
-				}
-				if vs.VolumeID == volUUID {
-					pes.envelopesToUpdate.Store(UUID, true)
-				}
-			}
-		} else {
-			pes.log.Errorf("No %v UUID in envelopes to check in affectedByVolumeStatus", UUID)
-		}
-	}
 }
 
 // UpdateEnvelopes sets pes.envelopes and marks envelopes that are not
 // present in new peInfo as ones to be deleted and updates the rest of them
 // all of the updates will happen after notification to updateStateNotificationCh
 // will be sent
 func (pes *PatchEnvelopes) UpdateEnvelopes(peInfo []types.PatchEnvelopeInfo) {
+	pes.Lock()
+	defer pes.Unlock()
```

Comment:
```suggestion
	pes.RLock()
	defer pes.RUnlock()
```
Because here you only read, don't you?

Otherwise you could just use sync.Mutex, but then the LockedMap would be useless.

Reply:
Fair

Reply:
Updated